chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredcitionServiceImpl.java

```diff
@@ -0,0 +1,18 @@
+package software.bytepushers.pick3.services;
+
+import com.amazonaws.services.lambda.AWSLambdaClientBuilder;
+import com.amazonaws.services.lambda.invoke.LambdaInvokerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class Pick3PredcitionServiceImpl implements Pick3PredictionService {
+
+    @Override
+    public int[][] generatePredictions(int winningNumber) {
+        final Pick3PredictionService pick3PredictionService = LambdaInvokerFactory.builder()
+                .lambdaClient(AWSLambdaClientBuilder.defaultClient())
+                .build(Pick3PredictionService.class);
```

Comment:
The LambdaInvoker is using the signature of methods defined in the `Pick3PredictionService` interface to construct an RPC handler. But the interface being used is the same interface that this service implements, which is weird.

A separate interface should be used for this purpose.

Reply:
Not sure I follow.